src/System.ObjectModel/tests/ReadOnlyDictionary/ReadOnlyDictionaryTests.cs

```diff
@@ -217,6 +218,186 @@ public static void DebuggerAttributeTests()
         }
     }
 
+    public class TestReadOnlyDictionary<TKey, TValue> : ReadOnlyDictionary<TKey, TValue>
+    {
+        public TestReadOnlyDictionary(IDictionary<TKey, TValue> dict)
+            : base(dict)
+        {
+        }
+
+        public IDictionary<TKey, TValue> GetDictionary()
+        {
+            return Dictionary;
+        }
+    }
+
+    public class DictionaryThatDoesntImplementNonGeneric<TKey, TValue> : IDictionary<TKey, TValue>
+    {
+        private readonly IDictionary<TKey, TValue> _inner;
+        
+        public DictionaryThatDoesntImplementNonGeneric(IDictionary<TKey, TValue> inner)
+        {
+            _inner = inner;
+        }
+
+        public IEnumerator<KeyValuePair<TKey, TValue>> GetEnumerator()
+        {
+            return _inner.GetEnumerator();
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return ((IEnumerable) _inner).GetEnumerator();
+        }
+
+        public void Add(KeyValuePair<TKey, TValue> item)
+        {
+            _inner.Add(item);
+        }
+
+        public void Clear()
+        {
+            _inner.Clear();
+        }
+
+        public bool Contains(KeyValuePair<TKey, TValue> item)
+        {
+            return _inner.Contains(item);
+        }
+
+        public void CopyTo(KeyValuePair<TKey, TValue>[] array, int arrayIndex)
+        {
+            _inner.CopyTo(array, arrayIndex);
+        }
+
+        public bool Remove(KeyValuePair<TKey, TValue> item)
+        {
+            return _inner.Remove(item);
+        }
+
+        public int Count
+        {
+            get { return _inner.Count; }
+        }
+
+        public bool IsReadOnly
+        {
+            get { return _inner.IsReadOnly; }
+        }
+
+        public void Add(TKey key, TValue value)
+        {
+            _inner.Add(key, value);
+        }
+
+        public bool ContainsKey(TKey key)
+        {
+            return _inner.ContainsKey(key);
+        }
+
+        public bool Remove(TKey key)
+        {
+            return _inner.Remove(key);
+        }
+
+        public bool TryGetValue(TKey key, out TValue value)
+        {
+            return _inner.TryGetValue(key, out value);
+        }
+
+        public TValue this[TKey key]
+        {
+            get { return _inner[key]; }
+            set { _inner[key] = value; }
+        }
+
+        public ICollection<TKey> Keys
+        {
+            get { return _inner.Keys; }
+        }
+
+        public ICollection<TValue> Values
+        {
+            get { return _inner.Values; }
+        }
+    }
+
+    public class ReadOnlyDictionaryOverNonGenericTests
+        : IDictionaryTest<string, int>
```

Comment:
DictionaryTests?  BaseIDictionaryTests? etc.  To a casual reader/reviewer, it just looks like the class is implementing an interface rather than deriving from a base class that actually exposes [Facts].